### PR TITLE
MAINT: Leverage SDCFlows 2.x API

### PR DESCRIPTION
**Pull request by mgxd**
## Changes proposed in this pull request

This PR decouples SDC workflows into a more modular structure, ala anatomical workflows via `smriprep`. A first pass of this integration was done in [`nibabies`](https://github.com/nipreps/nibabies/pull/43/files).
 
## Documentation that should be reviewed

- ~~TODO: Update documentation accordingly.~~

### CODE REVIEW

**Comment by oesteban** [comment]
Hey @mgxd, I was thinking about the issues you brought up in our meeting yesterday. One thing I did not realize is that we can (or should) check that the correction with SDCFlows and TOPUP on the reference image (the blip-up/down EPIs, not the target bold) are consistent. To run this check, just make sure `debug=True` when building the topup workflow (https://github.com/nipreps/sdcflows/blob/d2023f59a278a57cc7e15888fd545c349b8d82d2/sdcflows/workflows/fit/pepolar.py#L191-L220)

**Comment by oesteban** [comment]
Okay, I have rebased into current master and now I'm testing locally with the dataset you provided. I will report back as soon as that gets finished.

**Comment by oesteban** [comment]
Okay, let's proceed with the integration and treat the problem with the failing dataset as a bug (because it does look like a bug of SDCFlows, not a problem derived from the integration here).

I guess we want to test on more data from OpenNeuro that indeed has fieldmaps.

- ds031 could be a good benchmark to see the different fieldmaps (corresponding to sessions).
- what other datasets come to mind?

**Review by oesteban — COMMENTED** [review]
Shaping up - I will focus on the warping that SDCFlows is generating. When combined with head motion and other transforms, the final BOLD is looking pretty bad. I'll focus on that now, as I think it is the last hurdle before we get this merged.

File: fmriprep/interfaces/__init__.py
Comment (oesteban) [reply fmriprep/interfaces/__init__.py:25]: Keep an eye on #2495.

File: fmriprep/workflows/bold/base.py
Comment (oesteban) [reply fmriprep/workflows/bold/base.py]: Fixed in the commit it was introduced.

File: fmriprep/workflows/bold/base.py
Comment (oesteban) [reply fmriprep/workflows/bold/base.py]: Confirmed this was the reason.

**Comment by mgxd** [comment]
Okay, reports look decent enough - this is *finally* ready for review

File: fmriprep/workflows/base.py
Comment (effigies) [inline fmriprep/workflows/base.py:319]: Are we getting rid of SDC-SyN for 20.1?

Comment (mgxd) [reply fmriprep/workflows/base.py:319]: No, the plan is to include it in after this is merged (and we cut an RC)

Comment (effigies) [reply fmriprep/workflows/base.py:319]: Ah, okay.

**Comment by effigies** [comment]
Overall this looks reasonable. Some issues with the boilerplate, though:

fmriprep-sdcflows_boilerplate

FSL reference shows a Python object reference, and the anatomical header has been wrapped into the SDC section.

File: fmriprep/workflows/base.py
Comment (oesteban) [reply fmriprep/workflows/base.py:386]: bumping this suggestion up

**Review by oesteban — APPROVED** [review]

**Comment by oesteban** [comment]
> the anatomical header has been wrapped into the SDC section.

The easiest here would be to add a new line before the Anatomical section title when we set the `__desc__` of the workflow in *sMRIPrep*.

**Comment by effigies** [comment]
Why is fieldmap preceding anatomical, by the way? Is it easy to swap the order? Conceptually, I would put anatomical first.

**Comment by mgxd** [comment]
I think we are using `workflow.__desc__`/`workflow.__postdesc__` interchangeably. Will address for 21.0.0 release, but no reason to hold up the RC (related issue #2526)

**Comment by effigies** [comment]
Yup. I think this is good to go.

**Pull request merged**
